packages/core/src/Transform.ts

```diff
@@ -584,22 +581,75 @@ export class Transform extends Component {
     target._scale.copyFrom(this.scale);
   }
 
-  protected override _onDestroy(): void {
-    super._onDestroy();
-    //@ts-ignore
-    this._worldPosition._onValueChanged = null;
-    //@ts-ignore
-    this._rotation._onValueChanged = null;
-    //@ts-ignore
-    this._worldRotation._onValueChanged = null;
-    //@ts-ignore
-    this._rotationQuaternion._onValueChanged = null;
-    //@ts-ignore
-    this._worldRotationQuaternion._onValueChanged = null;
-    //@ts-ignore
-    this._position._onValueChanged = null;
-    //@ts-ignore
-    this._scale._onValueChanged = null;
+  protected _decomposeLocalMatrix(matrix: Matrix, pos: Vector3, quat: Quaternion, scale: Vector3): void {
+    matrix.decompose(pos, quat, scale);
+    this._setDirtyFlagTrue(TransformModifyFlags.LocalEuler);
+    this._setDirtyFlagFalse(TransformModifyFlags.LocalMatrix | TransformModifyFlags.LocalQuat);
+  }
+
+  protected _onWorldMatrixChange() {
+    this._setDirtyFlagFalse(TransformModifyFlags.WorldMatrix);
+  }
+
+  protected _isContainDirtyFlags(targetDirtyFlags: number): boolean {
+    return (this._dirtyFlag & targetDirtyFlags) === targetDirtyFlags;
+  }
+
+  protected _isContainDirtyFlag(type: number): boolean {
+    return (this._dirtyFlag & type) != 0;
+  }
+
+  protected _setDirtyFlagTrue(type: number) {
+    this._dirtyFlag |= type;
+  }
+
+  protected _setDirtyFlagFalse(type: number) {
+    this._dirtyFlag &= ~type;
+  }
+
+  protected _worldAssociatedChange(type: number): void {
```

Comment:
If the event is already dirty and dispatched, you avoid duplicate dispatch. Have you found some bugs or cases before?